### ISSUE TITLE
HDPI drawing issues

### ISSUE THREAD

**Issue by ltratt** [softdevteam/eco#101]
Eco doesn't seem to draw quite properly at HDPI resolutions (I'm running at 2560x1440):

![screen1](https://cloud.githubusercontent.com/assets/20318/13919275/c72da180-ef6c-11e5-9297-91e93022dc3b.png)

The main annoyance is that the font baseline is incorrect: note the "y" (and the brackets!) spill below the language box. It looks to me like the `y` offset of the language box is incorrect.

A minor annoyance is that the start and end brackets for the language box are indeed a pixel wide -- which at this resolution means that they're almost invisible.

**Comment by jaheba** [comment]
To add to this, on (at least) Mac it seems that font sizes other than multiples of 5 yields a strange rendering for selected text:

![longselection](https://cloud.githubusercontent.com/assets/133732/13925266/03dac9f8-ef88-11e5-869f-22992c06c487.jpg)

(Please note that the whole text is selected.)

**Comment by ltratt** [comment]
That problem I don't see. They may be related, of course.

**Comment by jaheba** [comment]
It only is visible if you have one really long token (or word).

**Comment by ltratt** [comment]
I've just tried a really long token, and I don't see that particular bug. That said, the fact that I see an incorrect y offset, and you see an incorrect width suggests either we (or Qt) have a bug with fonts at high resolutions.

**Comment by jaheba** [comment]
I found a fix for my issue.
The underlying problem is that the font size is measured in points and thus is in relation to inches. So setting the font size from `15` to `16` may result in the same width for a single character, say `10`. The width of 100 characters however is quite different, in my case `963` and `1068`.

Below is the code which sets the font width. If we set it to a more precise float value, the selection displays correctly.

``` python
class GlobalFont(object):
    def __init__(self, font, size):
        font = QtGui.QFont(font, size)
        self.setfont(font)

    def setfont(self, font):
        self.font = font
        self.fontm = QtGui.QFontMetrics(self.font)
        self.fontht = self.fontm.height() + 3
        self.fontwt = self.fontm.width(" ")
        #self.fontwt = self.fontm.width(" "*100)/100.0
```

The other issue may have to do with the 3 pixels offset, but that's only a wild guess.

**Comment by ltratt** [comment]
It looks like the `+3` hack only affects the line height, not the y offset within that line. It was worth trying!

I think the width thing is worth making a PR for, though.

**Comment by ptersilie** [comment]
Fixed in #113
